src/uu/ls/src/ls.rs

```diff
@@ -750,6 +747,81 @@ impl Config {
             }
         }
 
+        // According to ls info page, `--zero` implies the following flags:
+        //  - `--show-control-chars`
+        //  - `--format=single-column`
+        //  - `--color=none`
+        //  - `--quoting-style=literal`
+        // Current GNU ls implementation allows `--zero` Behavior to be
+        // overridden by later flags.
+        let zero_formats_opts = [
+            options::format::ACROSS,
+            options::format::COLUMNS,
+            options::format::COMMAS,
+            options::format::LONG,
+            options::format::LONG_NO_GROUP,
+            options::format::LONG_NO_OWNER,
+            options::format::LONG_NUMERIC_UID_GID,
+            options::format::ONE_LINE,
+            options::FORMAT,
+        ];
+        let zero_colors_opts = [options::COLOR];
+        let zero_show_control_opts = [options::HIDE_CONTROL_CHARS, options::SHOW_CONTROL_CHARS];
+        let zero_quoting_style_opts = [
+            options::QUOTING_STYLE,
+            options::quoting::C,
+            options::quoting::ESCAPE,
+            options::quoting::LITERAL,
+        ];
+        let get_last =
+            |flag: &str| -> usize { options.indices_of(flag).and_then(|x| x.last()).unwrap_or(0) };
```

Comment:
I think this can be simplified by simply using `index_of`. Because we don't allow the flag multiple times but we allow them to override themselves. This means that clap will completely disregard the earlier argument. See this playground link: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1e3cd4125a923537d4e65effccdd7d9d

Reply:
Also I think you can assume that only ever one argument will be present from each of those lists (again due to overriding).

Although I think that assumption does not hold for `LONG_NO_GROUP/OWNER`.

Reply:
Thanks, I now use this in 61f77e9b61812036b0b5804a2dee8d56cd28ef0c, also allowing multiple `--zero`, as GNU `ls` allows it.